main.go

```diff
@@ -358,23 +380,24 @@ func buildTargetDirForScheme(proj xcodeproj.XcodeProj, projectPath, scheme, conf
 			return "", fmt.Errorf("failed to parse workspace (%s) build settings, error: %s", projectPath, err)
 		}
 	} else {
-		return "", fmt.Errorf("project file extension should be .xcodeproj or .xcworkspace, but got: %s", ext)
+		return "", fmt.Errorf("project file extension should be .xcodeproj or .xcworkspace, but got: %s", filepath.Ext(projectPath))
+
 	}
 
-	schemeBuildDirObject := buildSettings["TARGET_BUILD_DIR"]
-	schemeBuildDir, ok := schemeBuildDirObject.(string)
-	if !ok {
-		return "", fmt.Errorf("failed to parse build settings")
+	schemeBuildDir, err := buildSettings.String("TARGET_BUILD_DIR")
+	if err != nil {
+		return "", fmt.Errorf("failed to parse build settings, error: %s", err)
 	}
 
 	return schemeBuildDir, nil
 }
 
 // exportArtifacts exports the main target and it's .app dependencies.
-func exportArtifacts(proj xcodeproj.XcodeProj, schemeBuildDir string, configuration, XcodebuildOptions, simulatorPlatform, deployDir string) error {
+func exportArtifacts(proj xcodeproj.XcodeProj, schemeBuildDir string, configuration, XcodebuildOptions, simulatorPlatform, deployDir string) ([]string, error) {
+	var exportedArtifacts []string
 	splitSchemeDir := strings.Split(schemeBuildDir, "Build/")
 	if len(splitSchemeDir) != 2 {
-		return fmt.Errorf("failed to parse scheme's build target dir: %s", schemeBuildDir)
+		return nil, fmt.Errorf("failed to parse scheme's build target dir: %s", schemeBuildDir)
 	}
 
 	for _, target := range proj.Proj.Targets {
```

Comment:
we are caring only about the scheme selected targets, what if the given scheme does not builds the watch target?